### ISSUE TITLE
Open app after user confirmation

### ISSUE THREAD

**Issue by kueda** [inaturalist/iNaturalistReactNative#761]
After a user has created their account, they need to click a link in the email we send them to confirm their account before they're allowed to sign in. When they click the link, they get taken to the website, but if they're on a device with the iNat mobile app installed, we want them to land in the app, not the website (presumably they signed up in the app).

To do this, we need to support [deep links](https://reactnative.dev/docs/linking#handling-deep-links) to intercept the following URLs:

```md
# This URL they land on after confirming a *new* account
https://www.inaturalist.org/users/sign_in?confirmed=true

# The URL they land on after confirming an existing account when they're already signed in on the website
https://www.inaturalist.org/home?confirmed=true
```

When the app intercepts either of these URLs, it needs to do the following:

- [x] Check if the user is signed in. If so, just land them on My Observations. If not, they land on the LogIn screen
- [x] The LogIn screen needs an additional message above the form that reads, "Your email is confirmed! Please log in to continue." [<img width="387" alt="Screenshot 2023-08-31 at 6 22 33 PM" src="https://github.com/inaturalist/iNaturalistReactNative/assets/22691/e3559ffb-ed84-47f9-916e-3a0cf92fd9b1">](https://www.figma.com/file/MChpvx4ZrKEVVwsWKt4lkI/iNaturalist-Mobile-UI-Design?type=design&node-id=3306-21058&mode=design&t=2PpFAVIg5C0NKyVC-0)

**Comment by albullington** [comment]
I believe for this to work we'll need an association file as described in the [iOS universal links documentation](https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html) as well as an asset file described in the [Android App Link documentation](https://developer.android.com/training/app-links/verify-android-applinks#web-assoc). Also, I haven't seen any documentation yet that makes me think React Native Linking can handle query strings, but this is hard to test since the links aren't working at the moment.

**Comment by albullington** [comment]
Closed via #782
